WordPress/src/main/java/org/wordpress/android/ui/accounts/login/MagicLinkSignInFragment.java

```diff
@@ -0,0 +1,251 @@
+            throw new RuntimeException(context.toString() + " must implement OnMagicLinkRequestInteraction");
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = super.onCreateView(inflater, container, savedInstanceState);
+
+        configureMagicLinkUI();
+        mUsernameEditText.setOnEditorActionListener(new TextView.OnEditorActionListener() {
+            @Override
+            public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
+                if (didPressNextKey(actionId, event) && !isEnterPasswordMode()) {
+                    signIn();
+                    return true;
+                } else {
+                    return false;
+                }
+            }
+        });
+
+        return view;
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+
+        if (!mToken.isEmpty()) {
+            attemptLoginWithMagicLink();
+        }
+        if (mShouldShowPassword) {
+            showPasswordFieldAndFocus();
+        }
+    }
+
+    public void setToken(String token) {
+        mToken = token;
+    }
+
+    public void setShouldShowPassword(boolean shouldShowPassword) {
+        mShouldShowPassword = shouldShowPassword;
+    }
+
+    @Override
+    protected void toggleSignInMode(){
+        if (mUrlButtonLayout.getVisibility() == View.VISIBLE) {
+            configureMagicLinkUI();
+            mSelfHosted = false;
+        } else {
+            showSelfHostedSignInForm();
+            mSelfHosted = true;
+        }
+    }
+
+    @Override
+    protected void signIn() {
+        if (mSelfHosted || isEnterPasswordMode()) {
+            super.signIn();
+        } else {
+            if (isUsernameEmail()) {
+                requestWPComEmailCheck();
+            } else {
+                showPasswordFieldAndFocus();
+            }
+        }
+    }
+
+    @Override
+    protected void track(AnalyticsTracker.Stat stat, Map<String, Boolean> properties) {
+        if (properties == null) {
+            properties = new HashMap<>();
+        }
+        properties.put(MAGIC_LINK_PROPERTY, true);
+        AnalyticsTracker.track(stat, properties);
+    }
+
+    @Override
+    protected void finishCurrentActivity(final List<Map<String, Object>> userBlogList) {
+        if (!isAdded()) {
+            return;
+        }
+        getActivity().runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                if (userBlogList != null) {
+                    Intent intent = new Intent(getActivity(), WPMainActivity.class);
+                    intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+                    intent.putExtra(MagicLinkSignInActivity.MAGIC_LOGIN, true);
+
+                    getActivity().startActivity(intent);
+                }
+            }
+        });
+    }
+
+    @Override
+    protected boolean isSmartLockAvailable() {
```

Comment:
Definitely temporary. We should not drop smart lock. Reason for this is because A/B testing results between this method and the previous one will be much easier to digest.